src/main/java/com/celements/cleverreach/DefaultFailNotificationHandler.java

```diff
@@ -0,0 +1,128 @@
+package com.celements.cleverreach;
+
+import static com.celements.model.util.References.*;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.model.reference.ClassReference;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.SpaceReference;
+
+import com.celements.cleverreach.exception.CleverReachRequestFailedException;
+import com.celements.cleverreach.exception.CssInlineException;
+import com.celements.common.classes.IClassCollectionRole;
+import com.celements.mailsender.IMailSenderRole;
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentNotExistsException;
+import com.celements.model.context.ModelContext;
+import com.celements.model.object.xwiki.XWikiObjectFetcher;
+import com.celements.web.classcollections.OldCoreClasses;
+import com.celements.web.classes.FormMailClass;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+@Component
+public class DefaultFailNotificationHandler implements FailNotificationHandlerRole {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(
+      DefaultFailNotificationHandler.class);
+
+  @Requirement
+  private ModelContext context;
+
+  @Requirement
+  private IModelAccessFacade modelAccess;
+
+  @Requirement
+  private IMailSenderRole mailSender;
+
+  @Requirement("celements.oldCoreClasses")
+  private IClassCollectionRole oldCoreClasses;
+
+  @Override
+  public void send(String msg, Exception excp) {
+    LOGGER.error(msg, excp);
+    try {
+      XWikiDocument configDoc = modelAccess.getDocument(getConfigDocRef());
+      List<BaseObject> receivers = XWikiObjectFetcher.on(configDoc).filter(
+          getReceiverEmailClassRef()).list();
+      String content = getMailingContent(msg, excp);
+      Optional<String> fromMail = getFromMail(configDoc);
+      for (BaseObject receiver : receivers) {
+        if (1 == receiver.getIntValue("is_active")) {
+          if (fromMail.isPresent()) {
+            mailSender.sendMail(fromMail.get(), null, receiver.getStringValue("email"), null,
+                null, "TAGESAGENDA UPDATE FAILED!", content, content, null, null);
+          } else {
+            LOGGER.error("Missing 'from' mail configuration for 'Tagesagenda update failed' "
+                + "notification");
+          }
+        }
+      }
```

Comment:
switch `for` and `if (fromMail.isPresent())` to avoid printing the same message multiple times.